Bottom: 170faa051757394d59b0b0377a7fb77b27fbed3b
Top:    dba4a9519931264c898197484aa2c359bbb3feae
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-25 02:14:07 +0000

Cure crosshair lagging on screen

FIXME: SHOULD NOT RE-WRITE THE GTK HID


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 26b1bb4..a1ea9ef 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -688,11 +688,12 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   static gint x_prev = -1, y_prev = -1;
   gboolean moved;
 
-
   gdk_event_request_motions (ev);
 
+#if 0
   if (!ghid_start_drawing (out))
     return FALSE;
+#endif
 
 
   if (out->panning)
@@ -712,11 +713,14 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   queue_tooltip_update (out);
 #endif
 
-  ghid_show_crosshair (TRUE);
-  if (moved && have_crosshair_attachments ())
+//  ghid_show_crosshair (TRUE);
+  if (moved) // && have_crosshair_attachments ())
     ghid_draw_area_update (gport, NULL);
 
+#if 0
   ghid_end_drawing (out);
+#endif
+
   return FALSE;
 }
